rct229/rules/section22/section22rule8.py

```diff
@@ -0,0 +1,141 @@
+        rmi_b = rmr_baseline["ruleset_model_instances"][0]
+        baseline_system_types_dict = get_baseline_system_types(rmi_b)
+        # create a list contains all HVAC systems that are modeled in the rmi_b
+        available_type_lists = [
+            hvac_type
+            for hvac_type in baseline_system_types_dict.keys()
+            if len(baseline_system_types_dict[hvac_type]) > 0
+        ]
+
+        primary_secondary_loop_dictionary = get_primary_secondary_loops_dict(rmi_b)
+
+        return (
+            any(
+                [
+                    available_type in APPLICABLE_SYS_TYPES
+                    for available_type in available_type_lists
+                ]
+            )
+            and primary_secondary_loop_dictionary
+        )
+
+    def create_data(self, context, data):
+        rmr_baseline = context.baseline
+        rmi_b = rmr_baseline["ruleset_model_instances"][0]
+        loop_pump_dictionary = {
+            pump["loop_or_piping"]: pump for pump in find_all("pumps[*]", rmi_b)
+        }
+
+        chw_loop_capacity_dict = {}
+        for chiller in find_all("chillers[*]", rmi_b):
+            if chiller["cooling_loop"] not in chw_loop_capacity_dict.keys():
+                chw_loop_capacity_dict[chiller["cooling_loop"]] = ZERO.POWER
+            chw_loop_capacity_dict[chiller["cooling_loop"]] += getattr_(
+                chiller, "chiller", "rated_capacity"
+            )
+
+        primary_secondary_loop_dictionary = get_primary_secondary_loops_dict(rmi_b)
+
+        child_loop_speed_control_dict = {}
+        for loop_b_id in primary_secondary_loop_dictionary.keys():
+            for child_loop in find_one_with_field_value(
+                "fluid_loops[*]", "id", loop_b_id, rmi_b
+            )["child_loops"]:
+                child_loop_speed_control_dict[child_loop["id"]] = loop_pump_dictionary[
+                    child_loop["id"]
+                ]["speed_control"]
+
+        return {
+            "loop_pump_dictionary": loop_pump_dictionary,
+            "chw_loop_capacity_dict": chw_loop_capacity_dict,
+            "child_loop_speed_control_dict": child_loop_speed_control_dict,
+        }
+
+    def list_filter(self, context_item, data):
+        fluid_loop_b = context_item.baseline
+        chw_loop_capacity_dict = data["chw_loop_capacity_dict"]
+        return (
+            fluid_loop_b["type"] == FLUID_LOOP.COOLING
+            and chw_loop_capacity_dict[fluid_loop_b["id"]]
+            >= MIN_CHW_PRIMARY_LOOP_COOLING_CAPACITY
+        )
+
+    class ChillerFluidLoopRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section22Rule8.ChillerFluidLoopRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                each_rule=Section22Rule8.ChillerFluidLoopRule.SecondaryChildLoopRule(),
+                index_rmr="baseline",
+                list_path="$..child_loops[*]",
+                required_fields={
```

Comment:
I don't think these required fields belong here.  This indicates that each chiller fluid loop must have a `type` field. But the rule doesn't use that field.

Reply:
Fixed. Great point!